plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/check/Expressions.kt

```diff
@@ -292,12 +296,26 @@ private fun SolverState.checkBlockExpression(
 ): ContSeq<StateAfter> =
   when {
     expressions.isEmpty() -> data.noReturn {}
+    // special case for finally { }
+    expressions.firstOrNull() is FinallySection -> {
+      val first = expressions.first() as FinallySection
+      val remainder = expressions.drop(1)
+      checkBlockExpression(associatedVarName, remainder, implicitReturnFromLast, data).onEach {
+        stateAfterRest ->
+        // override the return of the finally with the return of the try or catch
+        doOnlyWhenNotNull(first.finalExpression, data) { finalExpression ->
+          checkExpressionConstraintsWithNewName("finally", finalExpression, data).map {
+            stateAfterRest
+          }
+        }
+      }
+    }
     expressions.size == 1 &&
       implicitReturnFromLast -> // this is the last element, so it's the return value of the
       // expression
       checkExpressionConstraints(associatedVarName, expressions[0], data)
     else -> {
-      val first = expressions[0]
+      val first = expressions.first()
```

Comment:
Not sure why you changed this, but I've started preferring `first` over `[0]` so I like this change 😁 
I rarely ever find myself using indexed access, `first`, `tail`, `last`, `map`, etc cover all my use-cases.

Reply:
I didn't think that much, I just say `[0]` and thought "ugly!". I guess I'm slowly internalizing good Kotlin idioms :)